Headers/AppKit/NSCollectionViewLayout.h

```diff
@@ -0,0 +1,258 @@
+/* Interface of class NSCollectionViewLayout
+   Copyright (C) 2021 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: 30-05-2021
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#ifndef _NSCollectionViewLayout_h_GNUSTEP_GUI_INCLUDE
+#define _NSCollectionViewLayout_h_GNUSTEP_GUI_INCLUDE
+
+#import <Foundation/NSObject.h>
+#import <Foundation/NSGeometry.h>
+
+#import <AppKit/NSCollectionView.h>
+
+#if OS_API_VERSION(MAC_OS_X_VERSION_10_11, GS_API_LATEST)
+
+#if	defined(__cplusplus)
+extern "C" {
+#endif
+
+enum
+{
+ NSCollectionElementCategoryItem,
+ NSCollectionElementCategorySupplementaryView,
+ NSCollectionElementCategoryDecorationView,
+ NSCollectionElementCategoryInterItemGap
+}; 
+typedef NSInteger NSCollectionElementCategory;
+  
+
+typedef NSString *NSCollectionViewDecorationElementKind;
+NSCollectionViewSupplementaryElementKind NSCollectionElementKindInterItemGapIndicator;
+
+@class NSCollectionViewLayoutAttributes;
+@class NSCollectionView;
+@class NSDictionary;
+@class NSIndexPath;
+@class NSNib;
+@class NSSet;
+
+@interface NSCollectionViewLayoutAttributes : NSObject <NSCopying>
+{
+  NSRect _frame;
```

Comment:
Are you sure that these two values frame and size are completely independent of each other? I would have expected size to be the size of the frame.

Reply:
I changed this to use size in the _frame var.